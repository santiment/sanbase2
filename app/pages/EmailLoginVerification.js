import React from 'react'
import { connect } from 'react-redux'
import { compose, lifecycle } from 'recompose'
import * as qs from 'query-string'
import { Link } from 'react-router-dom'
import * as actions from './../actions/types'

export const EmailLoginVerification = ({ isSuccess, isError }) => {
  if (isError) {
    return (
      <div style={{ margin: '1em' }}>
        <h2>You do not have access.</h2>
        <p>Try again later. Maybe, your mail link is old.</p>
        <Link to='/login'>Login</Link>
      </div>
    )
  }
  if (isSuccess) {
    return (
      <div style={{ margin: '1em' }}>
        <h2>Email address confirmed</h2>
      </div>
    )
  }
  return (
    <div style={{ margin: '1em' }}>
      <h2>Verification...</h2>
    </div>
  )
}

const mapStateToProps = ({ rootUi }) => {
  return {
    isError: rootUi.loginError,
    isSuccess: rootUi.loginSuccess
  }
}

const mapDispatchToProps = dispatch => {
  return {
    emailLogin: payload => {
      dispatch({
        type: actions.USER_EMAIL_LOGIN,
        payload
      })
    }
  }
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  lifecycle({
    componentDidMount () {
      const payload = qs.parse(this.props.location.search)
      this.props.emailLogin(payload)
    }
  })
)

export default enhance(EmailLoginVerification)
