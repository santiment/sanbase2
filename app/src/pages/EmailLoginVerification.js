import React from 'react'
import { connect } from 'react-redux'
import {
  compose,
  withState,
  lifecycle
} from 'recompose'
import * as qs from 'query-string'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'

const emailLoginVerifyGQL = gql`
  mutation emailLoginVerify($email: String!, $token: String!) {
    emailLoginVerify(
      email: $email,
      token: $token
    ) {
      token
      user {
        id,
        email,
        username,
        ethAccounts {
          address,
          sanBalance
        }
      }
    }
  }
`

export const EmailLoginVerification = ({verificationStatus = 'pending'}) => {
  if (verificationStatus === 'pending') {
    return (
      <div>
        <h2>Verification...</h2>
      </div>
    )
  }
  if (verificationStatus === 'failed') {
    return (
      <div>
        <h2>You don't have access.</h2>
      </div>
    )
  }
  return (
    <div>
      <h2>Email address confirmed</h2>
    </div>
  )
}

const mapStateToProps = state => {
  return {
    user: state.user
  }
}

const mapDispatchToProps = dispatch => {
  return {
    authWithEmail: (qsData, props) => {
      props.verify(qsData)
        .then(({data}) => {
          const { token, user } = data.emailLoginVerify
          dispatch({
            type: 'SUCCESS_LOGIN',
            token,
            user
          })
          props.changeVerificationStatus('verified')
          props.history.push('/')
        })
        .catch(error => {
          dispatch({
            type: 'FAILED_LOGIN',
            errorMessage: error
          })
          props.changeVerificationStatus('failed')
        })
    }
  }
}

const enhance = compose(
  withState('verificationStatus', 'changeVerificationStatus', 'pending'),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  graphql(emailLoginVerifyGQL, {
    name: 'emailLoginVerify',
    props: ({ emailLoginVerify }) => ({
      verify: ({token, email}) => emailLoginVerify({ variables: { token, email } })
    }),
    options: { fetchPolicy: 'network-only' }
  }),
  lifecycle({
    componentDidMount () {
      const qsData = qs.parse(this.props.location.search)
      this.props.authWithEmail(qsData, this.props)
    }
  })
)

export default enhance(EmailLoginVerification)
