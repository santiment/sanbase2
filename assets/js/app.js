// js
import "phoenix_html"
import "./socket.js"

// css
import '../css/app.css';