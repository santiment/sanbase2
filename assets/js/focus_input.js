const FocusInput = {
  mounted() {

    document.addEventListener('keyup', (e) => {
      const input = document.getElementById('search-input');
      const search_result = document.getElementById('search-result-suggestions')
      if (document.activeElement != input) {
        if (e.key === '/') {
          input.focus();
          search_result.classList.remove('hidden')
        }
      }
    })
  }
}

export { FocusInput }
