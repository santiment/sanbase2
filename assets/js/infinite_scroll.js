export const InfiniteScroll = {
  mounted() {
    const observer = new IntersectionObserver(entries => {
      const entry = entries[0];
      if (entry.isIntersecting) {
        this.pushEvent('load_more');
      }
    }, {
      rootMargin: '200px',
    });

    observer.observe(this.el);
  },
  destroyed() {
    this.observer && this.observer.disconnect();
  }
}; 