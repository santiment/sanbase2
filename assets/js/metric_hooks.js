export const Sortable = {
  mounted() {
    const hook = this;
    
    new window.Sortable(this.el, {
      animation: 150,
      ghostClass: "bg-blue-100",
      onEnd: function(evt) {

        const ids = Array.from(hook.el.children).map(item => item.id);
        
        hook.pushEvent("reorder", { ids });
      }
    });
  }
};