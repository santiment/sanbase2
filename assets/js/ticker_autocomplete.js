export const TickerAutocomplete = {
  mounted() {
    this.handleEvent("ticker_suggestions", (payload) => {
      const { field, suggestions } = payload
      this.showSuggestions(field, suggestions)
    })
  },
  
  showSuggestions(field, suggestions) {
    // Find the suggestions container for this specific field
    // IDs are in format: base_asset_suggestions_1234 or quote_asset_suggestions_1234
    const container = document.querySelector(`[id^="${field}_suggestions_"]`)
    
    if (!container) return
    
    if (suggestions.length === 0) {
      container.classList.add('hidden')
      return
    }
    
    // Create suggestion items
    const suggestionItems = suggestions.map(ticker => 
      `<div class="px-3 py-2 hover:bg-gray-100 cursor-pointer text-sm border-b border-gray-100 last:border-b-0" data-ticker="${ticker}">
        ${ticker}
      </div>`
    ).join('')
    
    container.innerHTML = suggestionItems
    container.classList.remove('hidden')
    
    // Add click handlers for suggestions
    container.querySelectorAll('[data-ticker]').forEach(item => {
      item.addEventListener('click', (e) => {
        const ticker = e.target.getAttribute('data-ticker')
        // Find the corresponding input field by replacing _suggestions_ with _
        const inputId = container.id.replace('_suggestions_', '_')
        const input = document.getElementById(inputId)
        if (input) {
          input.value = ticker
          container.classList.add('hidden')
        }
      })
    })
  }
} 