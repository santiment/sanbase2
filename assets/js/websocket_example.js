
// import { Socket } from "phoenix"
let phoenix = require("phoenix")
let socket = new phoenix.Socket("wss://localhost:4000/socket", { params: {} })
socket.connect()

const channel = socket.channel("metrics:price", {})
channel.join()
    .receive('ok', () => { print("HELLO") })
    .receive('error', () => { print("Error") })
    .receive('timeout', () => { print("timeout") })