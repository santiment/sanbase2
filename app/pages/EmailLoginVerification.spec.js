/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { EmailLoginVerification } from './EmailLoginVerification.js'

describe('EmailLoginVerification', () => {
  it('it should render correctly', () => {
    const comp = shallow(
      <EmailLoginVerification
        location={{
          search: '?email=anyemail@sdf.com&token=adskfu2y9f2fejfhe'
        }}
      />
    )
    expect(toJson(comp)).toMatchSnapshot()
  })
})
