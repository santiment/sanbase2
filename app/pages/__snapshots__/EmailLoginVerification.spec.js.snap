// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailLoginVerification it should render correctly 1`] = `
<div
  style={
    Object {
      "margin": "1em",
    }
  }
>
  <h2>
    Verification...
  </h2>
</div>
`;
